wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -981,26 +1003,70 @@ public PushResponse push(PushRequest request) throws GitException, UnauthorizedE
 
             @SuppressWarnings("unchecked")
             Iterable<PushResult> list = (Iterable<PushResult>)executeRemoteCommand(remoteUri, pushCommand);
-            for (PushResult pushResult : list) {
-                Collection<RemoteRefUpdate> refUpdates = pushResult.getRemoteUpdates();
-                for (RemoteRefUpdate remoteRefUpdate : refUpdates) {
-                    if (!RemoteRefUpdate.Status.OK.equals(remoteRefUpdate.getStatus())) {
-                        if (RemoteRefUpdate.Status.UP_TO_DATE.equals(remoteRefUpdate.getStatus())) {
-                            return newDto(PushResponse.class).withCommandOutput("Everything up-to-date");
+            PushResult pushResult = list.iterator().next();
+            String currentBranch = getCurrentBranch();
+            return generatePushResponse(request, pushResult, currentBranch);
+        } catch (GitAPIException exception) {
+            return newDto(PushResponse.class).withCommandOutput(exception.getMessage());
+        }
+    }
+
+    private PushResponse generatePushResponse(final PushRequest request, final PushResult result, final String currentBranch) {
+        if (result != null) {
```

Comment:
ok so we tend to provide something null from something which is non null : the list (empty)
currently we have:
list: empty or not
get iterator which can be null
then check for null or not-null
null--> return PushResponse DTO
non null --> return PushResponse populated with commandOutput and updates

could it be more readable if :
we always generate PushResponse DTO
if list is not empty then we add commandOutput/updates


```
create dto
if list not empty
  addCommandAndOutput(dto, request, pushResult)
return dto
```

also I notice that currently we do
String currentBranch = getCurrentBranch();
generatePushResponse(request, pushResult, currentBranch);

we give currentBranch which is a class method so we could get rid of currentBranch parameter as generatePushResponse method is able to call it.